Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIBookmarksCocoa.mm

```diff
@@ -288,25 +163,26 @@
         return;
     }
 
-    std::vector<Ref<MockBookmarkNode>> allBookmarks;
-    allBookmarks.reserve(m_mockBookmarks.size());
-
-    for (const auto& pair : m_mockBookmarks)
-        allBookmarks.push_back(pair.value);
+    WebProcess::singleton().sendWithAsyncReply(
+        Messages::WebExtensionContext::BookmarksGetRecent(static_cast<uint64_t>(numberOfItems)),
+        [protectedThis = Ref { *this }, callback = WTFMove(callback)](Expected<Vector<WebExtensionBookmarksParameters>, WebExtensionError>&& result) {
+            if (!result) {
+                callback->reportError(result.error().createNSString().get());
+                return;
+            }
 
-    std::sort(allBookmarks.begin(), allBookmarks.end(), [](const Ref<MockBookmarkNode>& a, const Ref<MockBookmarkNode>& b) {
-        return a->dateAdded > b->dateAdded;
-    });
+            Vector<WebExtensionBookmarksParameters> recentBookmarksParams = result.value();
+            NSMutableArray *resultArray = [NSMutableArray arrayWithCapacity:recentBookmarksParams.size()];
 
-    NSMutableArray *resultArray = [NSMutableArray arrayWithCapacity:static_cast<NSUInteger>(numberOfItems)];
+            for (const auto& bookmarkParams : recentBookmarksParams) {
+                NSDictionary* bookmarkDictionary = toAPI(bookmarkParams);
+                [resultArray addObject:bookmarkDictionary];
```

Comment:
Super-nit: I think this could just be `[resultArray addObject:toAPI(bookmarkParams)];`